test/e2e/e2e_test.go

```diff
@@ -92,6 +106,15 @@ func TestE2E(t *testing.T) {
 	}
 
 	gomega.RegisterFailHandler(ginkgo.Fail)
+
+	// Ensure all pods are running and ready before starting tests (otherwise,
+	// cluster infrastructure pods that are being pulled or started can block
+	// test pods from running, and tests that ensure all pods are running and
+	// ready will fail).
+	if err := waitForPodsRunningReady(api.NamespaceDefault, minStartupPods, podStartupTimeout); err != nil {
```

Comment:
The comment is a little confusing, but what this function checks is that
- there is at least one pod
- that all pods are running and ready

The first condition catches the race in which this function runs before any pods are created; in that case, the second condition would be vacuously true.